R/internal.R

```diff
@@ -1,63 +1,75 @@
-                                               meta = NULL, 
-                                               script = "dist/react.min.js", 
-                                               stylesheet = NULL, 
-                                               head = NULL, 
-                                               attachment = NULL, 
-                                               package = "dashR", 
-                                               all_files = FALSE), 
-                                          class = "html_dependency"), 
-                        `react-prod` = structure(list(name = "react", 
-                                                      version = "16.2.0", 
-                                                      src = list(href = "https://unpkg.com/react@16.2.0", 
-                                                                 file = "lib/react@16.2.0"), 
-                                                      meta = NULL, 
-                                                      script = "umd/react.production.min.js", 
-                                                      stylesheet = NULL, 
-                                                      head = NULL, 
-                                                      attachment = NULL, 
-                                                      package = "dashR", 
-                                                      all_files = FALSE), 
-                                                 class = "html_dependency"), 
-                        `react-dom` = structure(list(name = "react-dom", 
-                                                     version = "15.4.2", 
+  deps_metadata <- list(`react` = structure(list(name = "react",
+                                               version = "15.4.2",
+                                               src = list(href = "https://unpkg.com/react@15.4.2",
+                                                          file = "lib/react@15.4.2"),
+                                               meta = NULL,
+                                               script = "dist/react.min.js",
+                                               stylesheet = NULL,
+                                               head = NULL,
+                                               attachment = NULL,
+                                               package = "dashR",
+                                               all_files = FALSE),
+                                          class = "html_dependency"),
+                        `react-prod` = structure(list(name = "react",
+                                                      version = "16.2.0",
+                                                      src = list(href = "https://unpkg.com/react@16.2.0",
+                                                                 file = "lib/react@16.2.0"),
+                                                      meta = NULL,
```

Comment:
It looks like this is just white space removal, but a few things strike me as odd about this code:
1. "react-prod" is react-16. This is different than the python side - we allow users to switch to react 16 optionally but it's not a "prod" vs "dev" distinction.
2. On the python side, we handle this in dash-renderer using the same structure as the rest of the component repos (https://github.com/plotly/dash-renderer/blob/1c7c75d3127a742093f59bd34ea23b9a84be1ac9/dash_renderer/__init__.py). I would be worried about DashR handling these versions independently of dash-renderer as we don't want DashPy to be distributed with a different version than DashR

This is non-blocking and not a high priority, but something to maybe break out into its own issue and consider down the line.

Reply:
Thanks for this comment -- this is among the last of the code/objects still retained from the "old" version of `dashR`. This function is written by the "transpiler" code when component libraries are generated, but I modify it directly as needed.

I'll open an issue to resolve this discrepancy and then link here.